Sources/TokenTextView/TokenTextView.swift

```diff
@@ -34,33 +24,49 @@ final class TokenTextView: UITextView {
     }
 
     // MARK: Private properties
+
     private var previousTextCount = 0
 
-    private(set) var tokenOpen: String = "{{"
-    private(set) var tokenClose: String = "}}"
+    private(set) var tokenOpen: String
+    private(set) var tokenClose: String
 
     private var tokenInstances = [TokenInstance]()
     private var pasteboardTokenInstances = [(TokenInstance, Int)]()
 
     // MARK: Setup
 
-    init(text: String? = nil, tokens: [Token] = [], tokenOpen: String = "{{", tokenClose: String = "}}") {
-        super.init(frame: .zero, textContainer: nil)
+    init(text: String? = nil,
+         tokenList: [Token] = [],
+         tokenOpen: String = "{{",
+         tokenClose: String = "}}",
+         tokenAttributes: Attributes = Attributes(backgroundColor: .purple, foregroundColor: .white, font: .systemFont(ofSize: 12.0)),
+         textAttributes: Attributes = Attributes(backgroundColor: .clear, foregroundColor: .black, font: .systemFont(ofSize: 12.0))) {
         self.tokenOpen = tokenOpen
         self.tokenClose = tokenClose
-        self.text = text
+        self.tokenAttributes = tokenAttributes
+        self.textAttributes = textAttributes
         defer {
-            self.tokens = tokens
+            self.tokenList = tokenList
         }
+        super.init(frame: .zero, textContainer: nil)
+        self.text = text
         setup()
     }
 
     override init(frame: CGRect, textContainer: NSTextContainer?) {
+        self.tokenOpen = "{{"
+        self.tokenClose = "}}"
+        self.tokenAttributes = Attributes(backgroundColor: .purple, foregroundColor: .white, font: .systemFont(ofSize: 12.0))
+        self.textAttributes = Attributes(backgroundColor: .clear, foregroundColor: .black, font: .systemFont(ofSize: 12.0))
         super.init(frame: frame, textContainer: textContainer)
         setup()
     }
 
     required init?(coder: NSCoder) {
+        self.tokenOpen = "{{"
+        self.tokenClose = "}}"
+        self.tokenAttributes = Attributes(backgroundColor: .purple, foregroundColor: .white, font: .systemFont(ofSize: 12.0))
+        self.textAttributes = Attributes(backgroundColor: .clear, foregroundColor: .black, font: .systemFont(ofSize: 12.0))
         super.init(coder: coder)
         setup()
     }
```

Comment:
seems like we are not enabling custom setup when using these inits, should we just disable the inits instead?

Reply:
I was able to remove the `override init(frame: CGRect, textContainer: NSTextContainer?)` [7e73800](https://github.com/open-turo/token-text-view/pull/15/commits/7e73800386f18e33be1d7ffe0ad379410a9796ff)